### PR TITLE
Fix blur-up preview image not working on initial page load

### LINKED ISSUE #14325: Blurred image only shown briefly
## Description

I was previously experiencing #14186. After upgrading to gatsby-remark-images 3.x the situation has improved, but is still not working correctly.

The blurred image only appears for a very short amount of time, and then is covered with a blank white rectangle.

### Expected result

When I load my page on a slow network connection, I expect the placeholder image to be displayed until the full image has loaded.

### Actual result

blurUp

(This GIF is a real-time recording using Firefox's GPRS network throttling preset)

### Environment

```

  System:
    OS: Windows 10
    CPU: (16) x64 AMD Ryzen 7 1800X Eight-Core Processor         
  Binaries:
    Yarn: 1.13.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.5.0-next.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 2.7.15
  Browsers:
    Edge: 44.17763.1.0
  npmPackages:
    gatsby: ^2.2.6 => 2.2.6 
    gatsby-image: ^2.0.22 => 2.0.25 
    gatsby-plugin-canonical-urls: ^2.0.8 => 2.0.8 
    gatsby-plugin-catch-links: ^2.0.9 => 2.0.9 
    gatsby-plugin-excerpts: ^1.0.2 => 1.0.2 
    gatsby-plugin-favicon: ^3.1.5 => 3.1.5 
    gatsby-plugin-feed: ^2.0.8 => 2.0.11 
    gatsby-plugin-google-analytics: ^2.0.5 => 2.0.8 
    gatsby-plugin-manifest: ^2.0.5 => 2.0.12 
    gatsby-plugin-meta-redirect: ^1.1.1 => 1.1.1 
    gatsby-plugin-offline: ^2.0.5 => 2.0.20 
    gatsby-plugin-prefetch-google-fonts: ^1.1.1 => 1.1.1 
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.5 
    gatsby-plugin-remove-serviceworker: ^1.0.0 => 1.0.0 
    gatsby-plugin-root-import: ^2.0.4 => 2.0.4 
    gatsby-plugin-s3: ^0.2.5 => 0.2.5 
    gatsby-plugin-sass: ^2.0.7 => 2.0.7 
    gatsby-plugin-sharp: ^2.0.30 => 2.0.30 
    gatsby-plugin-sitemap: ^2.0.4 => 2.0.4 
    gatsby-plugin-typography: ^2.2.0 => 2.2.3 
    gatsby-redirect-from: ^0.1.1 => 0.1.1 
    gatsby-remark-autolink-headers: ^2.0.12 => 2.0.12 
    gatsby-remark-behead: 0.0.3 => 0.0.3 
    gatsby-remark-copy-linked-files: ^2.0.5 => 2.0.8 
    gatsby-remark-custom-blocks: ^2.0.3 => 2.0.3 
    gatsby-remark-embed-youtube: 0.0.7 => 0.0.7 
    gatsby-remark-external-links: 0.0.4 => 0.0.4 
    gatsby-remark-images: ^3.0.0 => 3.0.12 
    gatsby-remark-katex: ^3.0.1 => 3.0.1 
    gatsby-remark-prismjs: ^3.0.0 => 3.1.4 
    gatsby-remark-responsive-iframe: ^2.0.5 => 2.0.8 
    gatsby-remark-smartypants: ^2.0.5 => 2.0.7 
    gatsby-source-filesystem: ^2.0.27 => 2.0.27 
    gatsby-transformer-remark: ^2.1.6 => 2.1.17 
    gatsby-transformer-sharp: ^2.1.3 => 2.1.9 

```

### PR DESCRIPTION
**Pull request by JoshuaWalsh**
## Description

This fixes a bug where the blur-up preview image would not work correctly on initial page load, because the image would load before the JavaScript necessary for the effect.

This change enables a simple blur-up effect (without the fade animation) to work even with JavaScript disabled. Once the JavaScript loads, the original fade effect is restored for any images that are still loading.

It's worth noting that this change does hide the alt text by setting the colour to transparent, and uses JavaScript to show the alt text once the image loading is completed (either due to failure or success) so for users with JavaScript disable the alt text will not appear. For this reason I have also made the alt text appear as the title of the image if the image didn't already have a title.

This change should not impact accessibility, as screen readers don't seem to care about the colour of image alt text.

Before:
blurUp

After:
gatsby-remark-images

## Related Issues

These changes fix #14325.

### CODE REVIEW

**Comment by JoshuaWalsh** [comment]
Hi @KyleAMathews , could you provide me some guidance here? I'm having trouble with the tests.

The gatsbygram e2e tests seem to be failing, but the gatsbygram example doesn't use gatsby-remark-images at all, so I can't see how I would've broken it. 

I'm also having trouble running the e2e tests locally.

**Comment by JoshuaWalsh** [comment]
Hey, I've updated my changes to match master, but now some of the tests are failing. I don't think the failures have anything to do with me, they seem transient. Could someone please re-run the CircleCI workflow?

File: packages/gatsby-remark-images/src/constants.js [hunk 1]
```diff
@@ -1,3 +1,13 @@
+exports.defaults = {
```
Comment (wardpeet) [inline packages/gatsby-remark-images/src/constants.js]: `exports.defaults` might be confused with `export default` so maybe rename it to something verbose like `DEFAULT_OPTIONS`.
```suggestion
exports.DEFAULT_OPTIONS= {
```

**Comment by JoshuaWalsh** [comment]
Thanks @wardpeet , checks are passing now. Looking forward to getting this merged! :)

**Pull request merged**